packages/embedded-provider-core/src/interfaces/platform.ts

```diff
@@ -2,10 +2,20 @@ import type { EmbeddedStorage } from "./storage";
 import type { AuthProvider } from "./auth";
 import type { URLParamsAccessor } from "./url-params";
 
+export interface Stamper {
+  init(): Promise<{ keyId: string; publicKey: string }>;
+  sign(data: string | Uint8Array | Buffer): Promise<string>;
+  stamp(payload: any): Promise<string>;
```

Comment:
Ideally, the payload should always be a Buffer.
Currently, we accept mixed types (string, object, Buffer), which is error-prone.

While JSON.stringify(data) may appear to produce the same output, behavior can vary. If any library serializes the data object differently from how Stamper originally signed it, the stamp will no longer be valid.